### PR TITLE
core[patch], langchain[patch]: fix required deps

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/langchain/tests/unit_tests/conftest.py [hunk 1]
```diff
@@ -59,7 +59,11 @@ def test_something():
                 # If we haven't yet checked whether the pkg is installed
                 # let's check it and store the result.
                 if pkg not in required_pkgs_info:
-                    required_pkgs_info[pkg] = util.find_spec(pkg) is not None
+                    try:
+                        installed = util.find_spec(pkg) is not None
```
[the remark this answers is not in the document]
Comment (baskaryan) [reply libs/langchain/tests/unit_tests/conftest.py:63]: was giving errors for "google" but let me try again, maybe was symptom of different issue

[the remark this answers is not in the document]
Comment (baskaryan) [reply libs/langchain/tests/unit_tests/conftest.py:63]: https://github.com/langchain-ai/langchain/actions/runs/7133277557/job/19425640725

File: libs/core/langchain_core/agents.py [hunk 2]
```diff
@@ -105,7 +105,7 @@ def messages(self) -> Sequence[BaseMessage]:
 
 
 def _convert_agent_action_to_messages(
-    agent_action: AgentAction
+    agent_action: AgentAction,
```
Comment (efriis) [inline libs/core/langchain_core/agents.py:108]: was linting messing up somehow?

File: libs/experimental/pyproject.toml [hunk 3]
```diff
@@ -17,19 +17,32 @@ presidio-analyzer = {version = "^2.2.33", optional = true}
 faker = {version = "^19.3.1", optional = true}
 vowpal-wabbit-next = {version = "0.6.0", optional = true}
 sentence-transformers = {version = "^2", optional = true}
+jinja2 = {version = "^3.1.2", optional = true}
```
Comment (efriis) [inline libs/experimental/pyproject.toml:20]: should we do `^3` to match langchain?

**Review by efriis — APPROVED** [review]
